src/js/map/data-layers.js

```diff
@@ -308,32 +310,67 @@ class DataLayers {
       this.layersData.push({configLayer, layer, data});
     }
     
-    // TODO: add a listener for the map to add/remove the layer based on zoom
-    // this.map.on('zoomend ', (e) => {
-    //   console.log('zoom = '+ this.map.getZoom());
-    //   if ( this.map.getZoom() > 7 ){ this.map.addLayer(layer)}
-    //   else if ( this.map.getZoom() <= 7 ){ this.map.removeLayer(layer)}
-    // });
+    //Listener to control the visibility zoom when zooming
+    this.map.on('zoomend ', (e) => {
+      if(configLayer.displayedFromZoomLevel){
+        //console.log('displayedFromZoomLevel = '+ configLayer.displayedFromZoomLevel);
+        //console.log('zoom = '+ this.map.getZoom());
+        if (cluster) {  
+          if (this.map.getZoom() >= configLayer.displayedFromZoomLevel && this.map.getZoom() < configLayer.displayedUpToZoomLevel){ 
+            this.map.addLayer(clusterLayer);
+          } else {
+            this.map.removeLayer(clusterLayer);
+          }
+        } else{
+          if (this.map.getZoom() >= configLayer.displayedFromZoomLevel && this.map.getZoom() < configLayer.displayedUpToZoomLevel){ 
+            this.map.addLayer(layer);
+          } else {
+            this.map.removeLayer(layer);
+          }
+        }
+        
+        
+      }
+    });
 
     // TODO: refactor showLayersOnLoad to showAllLayersOnLoad, it will be clearer
     if (this.mapConfig.showLayersOnLoad) {
-      if (cluster) {     
-        this.map.addLayer(clusterLayer);
+      if (cluster) {    
+        if(configLayer.displayedFromZoomLevel){
```

Comment:
Shouldn't it be 

if(configLayer.displayedFromZoomLevel or configLayer.displayedUpToZoomLevel)

?